psi4/driver/procrouting/proc.py

```diff
@@ -1228,10 +1228,31 @@ def scf_wavefunction_factory(name, ref_wfn, reference, **kwargs):
             wfn.set_sad_fitting_basissets(sad_fitting_list)
             optstash.restore()
 
-    # Deal with the EXTERN issues
+    if hasattr(core, "EXTERN") and 'external_potentials' in kwargs: 
+        core.print_out("\n  Warning! Both an external potential EXTERN object and the external_potential" +
+                       " keyword argument are specified. The external_potential keyword argument will be ignored.\n")
+
+    # If EXTERN is set, then place that potential on the wfn
     if hasattr(core, "EXTERN"):
+        wfn.set_external_potential_c(core.EXTERN) # This is for the FSAPT procedure
         wfn.set_external_potential(core.EXTERN)
 
+    elif 'external_potentials' in kwargs:
+        # For FSAPT, we can take a dictionary of external potentials, e.g.,
```

Comment:
An SCF wavefunction shouldn't have to know that it's part of an F/I-SAPT sequence which has an A, B, and C. An SCF wavefunction should only know about the external potential of its own Hamiltonian.

Reply:
Now we use the ```potentials_``` member variable.